test.sh

```diff
@@ -61,7 +61,7 @@ do
 done
 
 # Run tests in parallel up to $max_concurrent at a time.
-xargs -n 1 -a tests/my_tests -P $max_concurrent ./run_test.sh 
+parallel -n 1 -a tests/my_tests -P $max_concurrent ./run_test.sh 
```

Comment:
should we provide -j and # of jobs we want? By default it will be # of cores..

Reply:
-P should be doing the same. From the man page:
       --jobs N
       -j N
       --max-procs N
       -P N
           Number of jobslots on each machine. Run up to N jobs in parallel.  0 means as many
           as possible. Default is 100% which will run one job per CPU core on each machine.

Reply:
ah, TIL! Good to go.